### ISSUE TITLE
Tests for cross-builds are not working

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/rattler-build#393]
```yml
# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
context:
  name: test-package
  version: 0.1.0

outputs:
  - package:
      name: ${{ name }}
      version: ${{ version }}
    build:
      script:
        - touch $PREFIX/test.txt
    requirements:
      host:
        - python >=3.9
      run:
        - python >=3.9
    test:
      commands:
        - ls "$PREFIX" | grep test.txt
```

`rattler-build build -r conda-recipe/recipe.yaml` works but when doing cross-building (in my case on `osx-arm64` trying to build a `osx-64` package), I get the following error:

```bash
$ rattler-build build -r conda-recipe/recipe.yaml --target-platform=osx-64

Error:   × Failed to setup test environment: Cannot solve the request because of: No candidates were found for test-package ==0.1.0 h0dc7051_0.
  │ 
  ╰─▶ Cannot solve the request because of: No candidates were found for test-package ==0.1.0 h0dc7051_0.
```

**Comment by wolfv** [comment]
Thanks for filing this issue. We'll do some rewriting for the tests soon (to make them a list and use the newly proposed format. We'll keep this in mind and hopefully solve!

**Comment by wolfv** [comment]
This is fixed in #411
